### PR TITLE
8260589: Crash in JfrTraceIdLoadBarrier::load(_jclass*)

### PR DESCRIPTION
**Pull request by mgronlun**
Hi Denghui,

I did not manage to find a solution that would could reduce verbosity, but i did find one, although a bit involved, that lets us keep it local to and integrated in the core of JFR itself.

Can you do a test run with this suggestion to see if it works for you?

Thanks
Markus

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8260589](https://bugs.openjdk.java.net/browse/JDK-8260589): Crash in JfrTraceIdLoadBarrier::load(_jclass*)

### Reviewers
 * [Jaroslav Bachorik](https://openjdk.java.net/census#jbachorik) (@jbachorik - **Reviewer**)
 * [Erik Gahlin](https://openjdk.java.net/census#egahlin) (@egahlin - **Reviewer**)

### Contributors
 * Denghui Dong `<ddong@openjdk.org>`

### Download
To checkout this PR locally:
`$ git fetch https://git.openjdk.java.net/jdk pull/2738/head:pull/2738`
`$ git checkout pull/2738`

To update a local copy of the PR:
`$ git checkout pull/2738`
`$ git pull https://git.openjdk.java.net/jdk pull/2738/head`

### CODE REVIEW

File: src/hotspot/share/jfr/recorder/checkpoint/types/traceid/jfrTraceIdLoadBarrier.cpp [hunk 1]
```diff
@@ -71,10 +71,30 @@ void JfrTraceIdLoadBarrier::do_klasses(klass_callback callback, bool previous_ep
...
 traceid JfrTraceIdLoadBarrier::load(jclass jc) {
   assert(jc != NULL, "invariant");
   assert(JavaThread::current()->thread_state() == _thread_in_vm, "invariant");
-  const oop my_oop = JNIHandles::resolve(jc);
-  assert(my_oop != NULL, "invariant");
-  return load(java_lang_Class::as_Klass(my_oop));
+  const oop mirror = JNIHandles::resolve(jc);
+  assert(mirror != NULL, "invariant");
+  const Klass* const k = java_lang_Class::as_Klass(mirror);
+  return k != NULL ? load(k) : load_primitive(mirror);
```
Comment (D-D-H) [inline src/hotspot/share/jfr/recorder/checkpoint/types/traceid/jfrTraceIdLoadBarrier.cpp:99]: I think the same modification should be made in JfrTraceId::load_raw.

Comment (mgronlun) [reply src/hotspot/share/jfr/recorder/checkpoint/types/traceid/jfrTraceIdLoadBarrier.cpp:99]: Thanks, Denghui, I will make the same change there. I will also put you down as a contributor to this change set.

**Comment by D-D-H** [comment]
LGTM, although I'm not a reviewer :)

**Review by jbachorik — APPROVED** [review]

**Review by egahlin — APPROVED** [review]

**Comment by mgronlun** [comment]
Thanks for reviewing!

**Pull request closed without merging**